prowler/providers/aws/services/ec2/ec2_instance_not_directly_publicly_accessible_via_elb/ec2_instance_not_directly_publicly_accessible_via_elb.py

```diff
@@ -0,0 +1,31 @@
+from prowler.lib.check.models import Check, Check_Report_AWS
+from prowler.providers.aws.services.ec2.ec2_client import ec2_client
+from prowler.providers.aws.services.elb.elb_client import elb_client
+
+
+class ec2_instance_not_directly_publicly_accessible_via_elb(Check):
+    def execute(self):
+        findings = []
+
+        public_instances = {}
+        for lb in elb_client.loadbalancers:
+            if lb.public:
+                for instance in lb.instances_ids:
+                    public_instances[instance] = lb
```

Comment:
Here you should add more verifications before adding an instance to public, first you should ensure that exists a security group attached to the instance that has some inbound rules that make the instance public.

Please review the `ec2_securitygroup_allow_ingress_from_internet_to_all_ports` and `ec2_securitygroup_allow_ingress_from_internet_to_any_port` to ensure that the instance is really open to Internet and avoid false positives

Reply:
done